app/assets/javascripts/application.js

```diff
@@ -31,6 +32,8 @@ jQuery(function ($) {
 
   var $atomAutodiscoveryLink = $("link[type='application/atom+xml']").eq('0')
 
+  var $transitionSubscribeButton = document.querySelector("button[data-label='subscribe-button']")
```

Comment:
Is there only ever going to be one subscribe button on the page?

Reply:
Instead of using data label to find the button you can pass through a `js-` class in the component to use as a JavaScript hook. https://components.publishing.service.gov.uk/component-guide/button/with_js_classes

Reply:
Have yet to see a page that has any other. Unfortunately the button component doesn't accept an ID, although I could target the form?

Reply:
In fact that's what I've done.

Reply:
If it's not likely then I wouldn't worry about multiple subscribe buttons on the page until it's requested.

I'd use a `js-subscribe-button-event` class (or something similarly named) to target the element, and pass the class into the component. Then that class can be used with `querySelector()`.